app/src/main/java/org/oppia/android/app/topic/TopicFragmentPresenter.kt

```diff
@@ -70,24 +72,39 @@ class TopicFragmentPresenter @Inject constructor(
     viewModel.setTopicId(topicId)
     binding.viewModel = viewModel
 
-    setUpViewPager(viewPager, topicId)
+    setUpViewPager(viewPager2, topicId)
     return binding.root
   }
 
   private fun setCurrentTab(tab: TopicTab) {
-    viewPager.setCurrentItem(tab.ordinal, true)
+    viewPager2.setCurrentItem(tab.ordinal, true)
     logTopicEvents(tab)
   }
 
-  private fun setUpViewPager(viewPager: ViewPager, topicId: String) {
+  private fun setUpViewPager(viewPager2: ViewPager2, topicId: String) {
     val adapter =
-      ViewPagerAdapter(fragment.childFragmentManager, internalProfileId, topicId, storyId)
-    viewPager.adapter = adapter
-    tabLayout.setupWithViewPager(viewPager)
-    tabLayout.getTabAt(0)!!.setText(fragment.getString(R.string.info)).setIcon(tabIcons[0])
-    tabLayout.getTabAt(1)!!.setText(fragment.getString(R.string.lessons)).setIcon(tabIcons[1])
-    tabLayout.getTabAt(2)!!.setText(fragment.getString(R.string.practice)).setIcon(tabIcons[2])
-    tabLayout.getTabAt(3)!!.setText(fragment.getString(R.string.revision)).setIcon(tabIcons[3])
+      ViewPagerAdapter(fragment, internalProfileId, topicId, storyId)
+    viewPager2.adapter = adapter
+    TabLayoutMediator(tabLayout, viewPager2) { tab, position ->
+      when (position) {
+        0 -> {
+          tab.text = fragment.getString(R.string.info)
+          tab.icon = ContextCompat.getDrawable(fragment.context!!, tabIcons[0])
```

Comment:
Here & below: you can just use the activity as the context instead of ``fragment.context!!``

Reply:
done